std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,36 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayOrder[@sp A: Order]: Order[Array[A]] = new ArrayOrder[A]
```

Comment:
I think you want arrayEq and arrayPartialOrder for cases where you don't have a total Order.